File: 22101528_Azimli_Maryam.py
#initializing
Student1 = {
    "name": "Maryam",
    "age": 20,
    "department": "Computer Engineering"
}

#printing
print(Student1["department"])

#add a new element
Student1["year"]=3
print(Student1["year"])

#remove
del Student1["year"]
#print(Student1["year"])

#modify
Student1["department"]="CS"
print(Student1["department"])

#check existence of key
if "name" in Student1:
    print("Key exists")
else:
    print("Key doesnt exist")

#check existence of value
found=False
for value in Student1.values():
    if value == "Maryam":
        found=True
        break
if found:
    print("Value exists")
else:
    print("Value does not exist")

for key, value in Student1.items():
    print(key, value)

File: 22101528_Azimli_Maryam.rs
use std::collections::HashMap;
fn main() {
    //initialize
    let mut Student = HashMap::new();
    Student.insert("name", "Mary");
    Student.insert("age", "20");
    Student.insert("department", "CS");
    
    //printing
    println!("{}", Student["name"]);
    
    //adding new value
    Student.insert("year", "3");
    println!("{}", Student["year"]);
    
    //remove
    Student.remove("year");
    //println!("{}", Student["year"]);
    
    //modify
    Student.insert("department", "computer engineering");
    println!("{}", Student["department"]);
    
    //search for key
    if Student.contains_key("age") {
        println!("Key exists.");
    } else {
        println!("key doesnt exist");
    }
    
    // Check if the value exists in the HashMap
    let found = Student.values().any(|&value| value == "Mary");

    // Print the result
    if found {
        println!("Value exists.");
    } else {
        println!("Value does not exist");
    }
    
    //loop through
    for (key, value) in &Student {
        println!("{}: {}", key, value);
    }
}